### PR TITLE
Add annotations to osrm response including speed limits, unit and sign conventions

### PR DESCRIPTION
**Pull request by danpaz**
# Issue

Adds speed limiit (a.k.a maxspeed) annotations to the osrm response when requested. Also includes speedLimitUnit and speedLimitSign properties on the step to denote the commonly used sign style ([Vienna Convention](https://en.wikipedia.org/wiki/Vienna_Convention_on_Road_Signs_and_Signals) or [MUTCD](https://en.wikipedia.org/wiki/Manual_on_Uniform_Traffic_Control_Devices)) and display unit (km/h or mph) in each country/region.

This PR also adds distance, duration, and speed annotations to the response -- these shape attributes were already on the tripleg, this PR just serializes them for consistency with speed limits.

## Tasklist

 - [X] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [ ] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/tyr/route_serializer_osrm.cc [hunk 1]
```diff
@@ -180,6 +184,38 @@ std::unordered_map<std::string, std::string> iso2_to_iso3 =
      {"VI", "VIR"}, {"VN", "VNM"}, {"VU", "VUT"}, {"WF", "WLF"}, {"WS", "WSM"}, {"YE", "YEM"},
      {"YT", "MYT"}, {"ZA", "ZAF"}, {"ZM", "ZMB"}, {"ZW", "ZWE"}, {"CS", "SCG"}, {"AN", "ANT"}};
 
+// Sign style and unit conventions for speed limit signs by country.
+// Most countries use Vienna style and km/h, but the countries below
+// use MUTCD and/or mph conventions.
+std::unordered_map<std::string, std::pair<std::string, std::string>> speed_limit_info = {
+    {"AG", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
```
Comment (1ec5) [inline src/tyr/route_serializer_osrm.cc:191]: `AI` (Anguilla) also [uses](https://en.wikipedia.org/wiki/Road_signs_in_the_United_Kingdom#Virgin_Islands) mph with the Vienna sign.

File: src/tyr/route_serializer_osrm.cc [hunk 2]
```diff
@@ -180,6 +184,38 @@ std::unordered_map<std::string, std::string> iso2_to_iso3 =
...
+    {"AS", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"BS", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"BZ", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"CA", {kSpeedLimitSignMutcd, kSpeedLimitUnitsKph}},
+    {"DM", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"FK", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"GB", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"GD", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"GG", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
```
Comment (1ec5) [inline src/tyr/route_serializer_osrm.cc:200]: `GS` (South Georgia and South Sandwich Islands) also uses mph with the Vienna sign.

File: src/tyr/route_serializer_osrm.cc [hunk 3]
```diff
@@ -180,6 +184,38 @@ std::unordered_map<std::string, std::string> iso2_to_iso3 =
...
+    {"GU", {kSpeedLimitSignMutcd, kSpeedLimitUnitsMph}},
+    {"IM", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"JE", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"KN", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"KY", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"LC", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
+    {"LR", {kSpeedLimitSignMutcd, kSpeedLimitUnitsKph}},
+    {"MP", {kSpeedLimitSignVienna, kSpeedLimitUnitsMph}},
```
Comment (1ec5) [inline src/tyr/route_serializer_osrm.cc:208]: `MS` (Montserrat) also uses mph with the Vienna sign.

**Review by 1ec5 — COMMENTED** [review]
I left some errata based on changes I made to speed limit units in ideditor/country-coder#30. These are quite the edge cases, accounting for probably a couple dozen speed limit signs tops, so I don’t know if anyone cares to adjust them for OSRM-based endpoints of the Mapbox Directions API, but I figure it’s best to keep this additional list as correct as possible in case it gets propagated elsewhere in the future.

**Comment by danpaz** [comment]
> I left some errata based on changes I made to speed limit units in ideditor/country-coder#30.

@1ec5 thanks! Updated to include those.

File: proto/trip.proto [hunk 4]
```diff
@@ -303,6 +303,8 @@ message TripLeg {
     repeated uint32 time = 1 [packed=true]; // milliseconds
     repeated uint32 length = 2 [packed=true]; // decimeters
     repeated uint32 speed = 3 [packed=true]; // decimeters per sec
+    // 4 is reserved
+    repeated uint32 speed_limit = 5 [packed=true]; // speed limit in kph
```
[the remark this answers is not in the document]
Comment (danpaz) [reply proto/trip.proto:307]: Only kph is supported right now. See https://github.com/valhalla/valhalla/pull/2668/files#diff-f9a60bb2a19f43b74cce0b8bda2dcfb117dda58d7bbf2198eaf60fc9d3cdc596R415-R416

[the remark this answers is not in the document]
Comment (danpaz) [reply proto/trip.proto:307]: @dgearhart I pushed some tests in the gurka test file which includes some mph and kph examples.

**Review by dgearhart — APPROVED** [review]

**Review by dgearhart — COMMENTED** [review]
:ship:

**Review by dgearhart — APPROVED** [review]

**Pull request merged**